extensions/wikia/Email/Controller/PasswordResetLinkController.class.php

```diff
@@ -51,6 +53,25 @@ public function getSubject() {
 	 * @template passwordResetLink
 	 */
 	public function body() {
+		// PLATFORM-2762 - we need to log information about reset tokens
+		// this action should be triggered by anonymous user, so current user will most likely contain ip. as this
+		// will make the CheckUser logs harder to browse, we override currentUser with targetUser
+		$currentUser = $this->currentUser;
+		$targetUser = $this->targetUser;
+		if ( !$currentUser instanceof \User || !$currentUser->getId() ) {
+			if ( $targetUser instanceof \User && $targetUser->getId() ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)